Week 3/shopee-reduxtoolkit-typscript/src/components/Product.tsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+
+import { Link, useNavigate } from 'react-router-dom';
+import { useAppDispatch } from '../store/store';
+import { addToCart } from "../store/productSlice";
+import { faEye } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+
+const Product = ({ id, title, price, desc, image }: { id: String | undefined, title: String | undefined, price: Number, desc: String | undefined, image: String }) => {
+  
+  let quantity = 0;
+  const [open, setOpen] = React.useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  const [change,setChange]=React.useState(1);
+  const dispatch = useAppDispatch();
+  const handleAddToCart = (id: String | undefined, title: String | undefined, price: Number, desc: String | undefined, image: String) => {
+    console.log("product", id)
+    setChange(prev=>prev+1)
+    dispatch(addToCart({
+      id: String(id), title: String(title), price, desc: String(desc), image, quantity:change
+    }))
+  }
+  const navigate=useNavigate();
+  const handleRedirects=()=>{
+    navigate(`/detail-product`,{state:{id, title, price, desc, image,quantity:change}})
+    console.log(quantity)
+  }
+  return (
+    <div className="product__item">
+      <div className="item__card">
+        <img alt="green iguana" src={String(image)} style={{ width: "100%" }} />
```

Comment:
@phamhuutinh0605 không dùng String
```suggestion
        <img alt="green iguana" src={image} style={{ width: "100%" }} />
```